src/honeybadger/event_dispatch.cr

```diff
@@ -0,0 +1,102 @@
+require "./event"
+
+module Honeybadger
+  class EventDispatch
+    # Allow for a spike of events
+    @channel = Channel(Event).new(1 << 16)
+    @buffer = IO::Memory.new
+
+    getter? running = false
+
+    def send(event : Event)
+      ensure_running
+
+      select
+      when @channel.send event
+      else # if the buffer is full, we just skip
+      end
+    end
+
+    private def ensure_running
+      sync { start }
+    end
+
+    private def start
+      return if running?
+
+      @running = true
+      spawn run
+
+      at_exit do
+        # Give the consumer fiber a chance to buffer any further events
+        Fiber.yield
+        send @buffer.to_s
+      end
+    end
+
+    private def run
+      next_event = nil
+
+      # Main run loop
+      # 1. Buffer messages until either 60 seconds have passed or the buffer
+      #    exceeds 5MB
+      # 2. Flush the buffer to the Honeybadger API
+      # 3. Clear the buffer
+      loop do
+        buffering = true
+        wait_time = 60.seconds
```

Comment:
Should this be configurable?

Reply:
I set it based on feedback in [this comment](https://github.com/honeybadger-io/honeybadger-crystal/pull/37#issuecomment-2240271335). It was originally sending every second.